sympy/solvers/tests/test_ode.py

```diff
@@ -3029,12 +3044,61 @@ def test_2nd_power_series_regular():
         + O(x**6))
 
     eq = x**2*(f(x).diff(x, 2)) + x*(f(x).diff(x)) + (x**2 - S(1)/4)*f(x)
-    assert dsolve(eq) == Eq(f(x), C1*(x**4/24 - x**2/2 + 1)/sqrt(x) +
+    assert dsolve(eq, hint='2nd_power_series_regular') == Eq(f(x), C1*(x**4/24 - x**2/2 + 1)/sqrt(x) +
         C2*sqrt(x)*(x**4/120 - x**2/6 + 1) + O(x**6))
 
     eq = x*(f(x).diff(x, 2)) - f(x).diff(x) + 4*x**3*f(x)
     assert dsolve(eq) == Eq(f(x), C2*(-x**4/2 + 1) + C1*x**2 + O(x**6))
 
+def test_2nd_linear_bessel_equation():
+    eq = x**2*(f(x).diff(x, 2)) + x*(f(x).diff(x)) + (x**2 - 4)*f(x)
+    sol = Eq(f(x), C1*besselj(2, x) + C2*bessely(2, x))
+    sols = constant_renumber(sol)
+    assert classify_ode(eq) == ('2nd_linear_bessel', '2nd_power_series_regular')
+    assert dsolve(eq, f(x)) in (sol, sols)
+    assert dsolve(eq, f(x), hint='2nd_linear_bessel') in (sol, sols)
+    assert checkodesol(eq, sol, order=2, solve_for_func=False) == (True, 0)
+
+    eq = x**2*(f(x).diff(x, 2)) + x*(f(x).diff(x)) + (x**2 +25)*f(x)
+    sol = Eq(f(x), C1*besselj(5*I, x) + C2*bessely(5*I, x))
+    sols = constant_renumber(sol)
+    assert classify_ode(eq) == ('2nd_linear_bessel', '2nd_power_series_regular')
+    assert dsolve(eq, f(x)) in (sol, sols)
+    assert dsolve(eq, f(x), hint='2nd_linear_bessel') in (sol, sols)
+    # FIXME: assert checkodesol(eq, sol, order=2, solve_for_func=False) == (True, 0)
```

Comment:
Would checkodesol work here as well using #16688?

Reply:
It shows the error of complex comparison during sorting(`for ba in sorted(besseljs, key=lambda x: x.args[0]):`).

```
def bessel_simp(expr):
    while True:
        besseljs = expr.find(lambda x: isinstance(x, besselj))
        for ba in sorted(besseljs, key=lambda x: x.args[0]):
            a, x = ba.args
            bap1 = besselj(a+1, x)
            bap2 = besselj(a+2, x)
            if expr.has(bap1) and expr.has(bap2):
                expr = expr.subs(ba, 2*(a+1)/x*bap1 - bap2)
                break
        else:
            return expr
```

Reply:
Change it ro `re(x.args[0])`.

Reply:
yes now it is working by using` re(args[0])`